Set Rounded Corners on Rectangle

seanlynchwv opened the pull request:
### Description

It's now possible to set the radius to create rounded corners on rectangle shapes.

```
        $panel = new AutoShape();
        $panel->setType(AutoShape::TYPE_ROUNDED_RECTANGLE)
            ->setOffsetX($x)
            ->setOffsetY($y)
            ->setWidth($w)
            ->setHeight($h)
            ->setRoundRectCorner(4); // new
```

Previously the corners were hard coded to a large value making custom designs difficult.

Fixes # (issue)
https://github.com/PHPOffice/PHPPresentation/issues/854

### Checklist:

- [x] My CI is :green_circle:
- [x] I have covered by unit tests my new code (check build/coverage for coverage report)
- [ ] I have updated the [documentation](https://github.com/PHPOffice/PHPPresentation/tree/develop/docs) to describe the changes
- [x] I have updated the [changelog](https://github.com/PHPOffice/PHPPresentation/blob/develop/docs/changes/1.1.0.md)

seanlynchwv committed:
set rounded corners

seanlynchwv committed:
set type

seanlynchwv committed:
update type

seanlynchwv committed:
pass cs-fixer using static assertions

seanlynchwv committed:
fix cs-fixer

seanlynchwv committed:
fix cs-fixer

seanlynchwv committed:
fix phpstan

seanlynchwv committed:
fix formatting

seanlynchwv committed:
attempt fix php8.4

seanlynchwv committed:
update changelog 1.2.0

Progi1984 commented on docs/changes/1.2.0.md:
```diff
@@ -17,6 +17,7 @@
 - `phpoffice/phpspreadsheet`: Allow version 4.0 by [@nreynis](https://github.com/nreynis) in [#861](https://github.com/PHPOffice/PHPPresentation/pull/861)
 - Smaller package size by [@nreynis](https://github.com/nreynis) in [#862](https://github.com/PHPOffice/PHPPresentation/pull/862)
 - Added setFirstSliceAngle to Doughnut Chart by [@seanlynchwv](http://github.com/seanlynchwv) in [#872](https://github.com/PHPOffice/PHPPresentation/pull/872)
+- Added ability to set rounded corner radius by [@seanlynchwv](http://github.com/seanlynchwv) in [#880](https://github.com/PHPOffice/PHPPresentation/pull/880)
```
@seanlynchwv Could you move your change to 1.3.0.md, please ?

Progi1984 commented on src/PhpPresentation/Shape/AutoShape.php:
```diff
@@ -273,4 +273,31 @@ public function setOutline(Outline $outline): self
     }
+
+    /** @var null|int */
+    private $roundRectAdj;
```
```suggestion
    protected $roundRectAdj;
```

Progi1984 commented on src/PhpPresentation/Shape/AutoShape.php:
```diff
@@ -273,4 +273,31 @@ public function setOutline(Outline $outline): self
+    {
+        $minHalf = (int) floor(min($this->width, $this->height) / 2);
+        if ($minHalf > 0) {
+            $this->roundRectAdj = max(0, min(50000, (int) round($pixels / $minHalf * 50000)));
```
Where does come from 50000 ?

Progi1984 replied:
You must move this calcul on the writer side, not on the autoshape class.

seanlynchwv replied:
Done! According to ChatGPT, the 50,000 comes from `the OOXML-defined max adjustment for the rounded-rectangle corner handle`

Progi1984 commented on tests/PhpPresentation/Tests/Shape/AutoShapeTest.php:
```diff
@@ -64,4 +69,83 @@ public function testType(): void
         self::assertEquals(AutoShape::TYPE_HEXAGON, $object->getType());
     }
+
+    public function testPixelSetterComputesAdjAndAffectsHash(): void
```
Move this test to https://github.com/PHPOffice/PHPPresentation/blob/master/tests%2FPhpPresentation%2FTests%2FWriter%2FPowerPoint2007%2FPptSlidesTest.php

Progi1984 reviewed (changes requested):
Some feedbacks

seanlynchwv committed:
Update src/PhpPresentation/Shape/AutoShape.php

Co-authored-by: Progi1984 <progi1984@gmail.com>

seanlynchwv committed:
refactor writer, px

seanlynchwv committed:
fix conflict

seanlynchwv committed:
Merge remote-tracking branch 'upstream/master' into feature/set-rounded-rect-corners

seanlynchwv committed:
merge master, update 1.3.0.md

seanlynchwv committed:
fix

seanlynchwv commented:
Hey @Progi1984 

CI passes on PHP 7.1 and 7.4+ but fails on 7.2–7.3 because, as far as I can tell, the allowed phpoffice/phpspreadsheet versions (^1.9–^5.0) no longer include any release compatible with PHP 7.2/7.3.

For the next major release `1.3.0`, could we set php 7.4 as the new minimum? PHP 7.4 reached EOL on November 28, 2022.

seanlynchwv committed:
Merge branch 'master' into feature/set-rounded-rect-corners

Progi1984 reviewed (commented):
A small feedback

seanlynchwv committed:
Update src/PhpPresentation/Shape/AutoShape.php

Co-authored-by: Progi1984 <progi1984@gmail.com>

seanlynchwv committed:
Move roundRect adj guide test into PptSlidesTest

Relocates testWriterEmitsAdjGuideForRoundRect from AutoShapeTest to
PptSlidesTest per maintainer feedback on PR #880, using the shared
zip/XML assertion helpers instead of manual ZipArchive/regex parsing.

seanlynchwv committed:
move test
